### PR TITLE
Initializing GlimmerLibUI DisplayService

### PR DESCRIPTION
**Pull request by Schwad**
This creates the initial structure to build off the GlimmerLibUI library as a displayservice. Glimmer also supports tk and some other things so I've namespaced this one specifically.

I just want to sanity check that I have the right things in the right directories to provide an example and inspiration for others jumping into displayservices; and also get started. The guts are still all unedited webview and I will break down the pieces as I look into this

cc @noahgibbs does this look right or am I missing something from a file structure shell perspective?

### CODE REVIEW

File: lib/scarpe/display_service.rb [hunk 1]
```diff
@@ -80,7 +80,10 @@ def subscribe_to_event(event_type, event_name, event_target, &handler)
 
       # TODO: add more display service types, use an env var to switch
       def display_services
-        @service_list ||= [WebviewDisplayService.new]
+        @service_list ||= [
+          WebviewDisplayService.new,
+          GlimmerLibUIDisplayService.new,
```
Comment (noahgibbs) [inline lib/scarpe/display_service.rb:85]: If you do this, it's going to create *both* Webview and Glimmer. As long as that's intended, cool. You can see my other PR for how to do this with an env var.

File: lib/scarpe/display_service.rb [hunk 2]
```diff
@@ -181,4 +184,74 @@ def query_display_widget_for(id, nil_ok: false)
       display_widget
     end
   end
+
+  class GlimmerDisplayService
```
Comment (noahgibbs) [inline lib/scarpe/display_service.rb:188]: I'm gonna extract the WebviewDisplayService into another file in the webview directory. But for now, this is great.

Comment (noahgibbs) [reply lib/scarpe/display_service.rb:188]: Also, a lot of this is still doing it the Webview way (e.g. it says to create WebviewApp and it's looking up class names for Webview widgets.) So I'm assuming this is just an initial copy-paste? In the final version you'll want to look up the Glimmer stuff, no Webview.

File: lib/scarpe/glibui/app.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      # SCARPE_TEST_CONTROL to its file path. This can
+      # allow pre-setting test options or otherwise
+      # performing additional actions not written into
+      # the Shoes app itself.
+      #
+      # The control interface is what lets these files see
+      # events, specify overrides and so on.
+      @control_interface = ControlInterface.new
+      if ENV["SCARPE_TEST_CONTROL"]
+        @control_interface.instance_eval File.read(ENV["SCARPE_TEST_CONTROL"])
```
Comment (noahgibbs) [inline lib/scarpe/glibui/app.rb:27]: You can decide whether the control interface is a thing you want here. It adds a fair bit of complexity, though it *does* make evented testing much easier.

File: lib/scarpe/glibui/app.rb [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      end
+
+      # TODO: rename @view
+      @view = Scarpe::WebWrangler.new title: @title,
```
Comment (noahgibbs) [inline lib/scarpe/glibui/app.rb:31]: I don't think you should need a WebWrangler for Glimmer.

**Comment by Schwad** [comment]
I'm back on the scene. Cleaning up as best I can. I MIGHT have this only turn on with an env var for now and then maybe fix forward a better (more elegant) way to do different display services. Probably a --option thingy

**Comment by Schwad** [comment]
Oh damn I'm wrong @noahgibbs I see we have `ENV["SCARPE_DISPLAY_SERVICES"]` on rebase good stuff!!!

**Comment by Schwad** [comment]
Now I'm just noodling through the errors after the rebase which is helping me in cleaning things up a bit. 

then I'll give it an eyeball. for this PR I want it as close to "hello world" feature-wise as possible. Then lots of teeny PRs after to grow it. :) 

Really like how it "puts" out the glimmer code when it runs. great feature.

File: lib/scarpe/glibui/alert.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
```
Comment (Schwad) [inline lib/scarpe/glibui/alert.rb:2]: we can ignore everything but para for this pr. but I'm keeping these as prompts for the next prs.

File: lib/scarpe/glibui/document_root.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+      }.show
+      GTEXT
+    end
+
+    def element(&blck)
+      window(&blck).show
+    end
+
+    # Bind a Scarpe callback name; see Scarpe::Widget for how the naming is set up
+    def bind(name, &block)
```
Comment (Schwad) [inline lib/scarpe/glibui/document_root.rb:31]: Not sure if bind needs to leave altogether or if this one is wanted generally

cc @noahgibbs

Comment (noahgibbs) [reply lib/scarpe/glibui/document_root.rb:31]: If you're going to convert Scarpe handlers into Glimmer handlers and handle them 100% in Glimmer then Scarpe doesn't need bind. I think that may be easier said than done, though.

Comment (Schwad) [reply lib/scarpe/glibui/document_root.rb:30]: Thanks for the context. Let's keep this in for now. Any changes can be separate issues.

**Pull request merged**